### PR TITLE
Add support for multiple preference types

### PR DESCRIPTION
**Pull request by ZvonimirMedak**
## Summary

Based on #32 I have added support for multiple preference types in SwiftUI

### Type

- [X] **Feature**: This pull request introduces a new feature.
- [ ] **Bug fix**: This pull request fixes a bug.
- [X] **Refactor**: This pull request refactors existing code.
- [X] **Documentation**: This pull request updates documentation.
- [ ] **Other**: This pull request makes other changes.

#### Additional information

- [ ] This pull request introduces a **breaking change**.

### Description

This PR adds a PreferenceItem protocol which will be used on the Preferences screen.
Added the support for Text, Int, Float, Double, and picker (CustomStringConvertible + RawRepresentable).
There are also validators which can be used with those items. 
The UI has also been updated to have a description label below the item which will show the description or the last error message from the Validator

## Checklist

- [X] I have performed a self-review of my own code.
- [X] I have tested my changes, including edge cases.
- [X] I have added necessary tests for the changes introduced (if applicable).
- [X] I have updated the documentation to reflect my changes (if applicable).

## Additional notes

The ToggleToolItem kept the same name due to it already existing in Sentinel, maybe we can create a type alias which would be PreferenceBoolItem, and then we could use both without creating a breaking change.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -176,7 +176,79 @@ The `tools` object is an array of `Tool` objects. `Tool` objects represent tools
 
 #### Preferences
 
-Last, but not the least, is the `preferences` object which is an array of `PreferencesTool.Section` objects. `PreferencesTool.Section` is used to allow the user to create sections of toggleable options. Each section consists of `ToggleToolItem` objects. To allow the user the interaction, you will have to add a `getter`, and a `setter` for the property from the `UserDefaults` object.
+Last, but not the least, is the `preferences` object which is an array of `PreferencesTool.Section` objects. `PreferencesTool.Section` is used to allow the user to create sections of editable options. Each section consists of `PreferenceItem` objects.
```
Comment (nikolamajcen) [inline README.md:179]: I'm fine with this, but `README.md` is getting bigger and bigger. I would put this here for now, but if the file will grow a bit more, in future, I would recommend creating a file under `docs` folder and link it in the `README.md` file.

File: Sentinel/Classes/Core/PreferencesInfo/Items/ToggleToolItem.swift [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+//  ToggleToolItem.swift
+//  Sentinel
+//
+//  Created by Zvonimir Medak on 17.03.2025..
+//
+
+import Foundation
+
+/// Item which will create a preferences view with a toggle button
+public struct ToggleToolItem: PreferenceItem {
```
Comment (nikolamajcen) [inline Sentinel/Classes/Core/PreferencesInfo/Items/ToggleToolItem.swift:11]: Every other item has a prefix `Preference`, so I would recommend going with something like `PreferenceToggleItem`.

You can add a typealias `ToggleToolItem` for `PreferenceToggleItem` and add a deprecation message for it, to still support that type from the "outside".

File: Sentinel/Classes/Core/PreferencesInfo/Views/OptionToggleView.swift [hunk 3]
```diff
@@ -0,0 +1,58 @@
+//
+//  OptionToggleView.swift
+//  Sentinel
+//
+//  Created by Zvonimir Medak on 27.11.2024..
+//
+
+import SwiftUI
+
+struct OptionToggleView: View {
```
Comment (nikolamajcen) [inline Sentinel/Classes/Core/PreferencesInfo/Views/OptionToggleView.swift:10]:
```suggestion
struct PreferenceToggleView: View {
```

File: Sentinel/Classes/Core/PreferencesInfo/Views/OptionToggleView.swift [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+                errorMessage = nil
+                onValueChanged(value)
+                return
+            }
+            errorMessage = message
+        }
+        .onAppear {
+            let fetchedValue = getter()
+            guard fetchedValue != value else { return }
+            value = fetchedValue
```
Comment (nikolamajcen) [inline Sentinel/Classes/Core/PreferencesInfo/Views/OptionToggleView.swift:41]: Can this be changed into something like this?
```swift
guard let fetchedValue = getter() else { return }
value = fetchedValue
```

File: Sentinel/Classes/Core/PreferencesInfo/Views/OptionToggleView.swift [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+
+            DescriptionView(description: description, errorMessage: errorMessage)
+        }
+        .onChange(of: value) { value in
+            guard let message = hasErrorMessage(value) else {
+                errorMessage = nil
+                onValueChanged(value)
+                return
+            }
+            errorMessage = message
```
Comment (nikolamajcen) [inline Sentinel/Classes/Core/PreferencesInfo/Views/OptionToggleView.swift:36]: I'm fine with this, but this can also be extracted into a separate method with descriptive method name.

File: Sentinel/Classes/Core/PreferencesInfo/Views/PreferencesView.swift [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+//  Pods
+//
+//  Created by Zvonimir Medak on 17.03.2025..
+//
+
+import SwiftUI
+
+struct PreferencesView: View {
+
+    let preferenceItem: any PreferenceItem
```
Comment (nikolamajcen) [inline Sentinel/Classes/Core/PreferencesInfo/Views/PreferencesView.swift:12]: To make the naming more consistent throughout the library.
```suggestion
    let item: any PreferenceItem
```

File: Sentinel/Classes/Core/PreferencesInfo/Views/PreferenceTextView.swift [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+                errorMessage = nil
+                onValueChanged(value)
+                return
+            }
+            errorMessage = message
+        }
+        .onAppear {
+            let fetchedValue = getter()
+            guard fetchedValue != value else { return }
+            value = fetchedValue
```
Comment (nikolamajcen) [inline Sentinel/Classes/Core/PreferencesInfo/Views/PreferenceTextView.swift:44]: Same comment as above.

File: Sentinel/Documentation.docc/Articles/Setup.md [hunk 8]
```diff
@@ -110,7 +110,81 @@ public static func createSentinelView(with configuration: Configuration) -> Sent
 
 ## Add Preferences
 
-The `prefernces` can be added as part of the configuration setup of **Sentinel**. Prefrences add items to the Preferences tab of Sentinel which are toggleable flags. Preference sections can be added:
+The `prefernces` can be added as part of the configuration setup of **Sentinel**. Prefrences add items to the Preferences tab of Sentinel which are editable items. Each section consists of `PreferenceItem` objects.
```
Comment (nikolamajcen) [inline Sentinel/Documentation.docc/Articles/Setup.md:113]: `prefernces` typo.

**Review by nikolamajcen — CHANGES_REQUESTED** [review]
Great work @ZvonimirMedak 👏  I've added couple of comments where I think this can be improved. Feel free to add your feedback, or let me know if you need additional support.

File: README.md (hunk 1 above)
Comment (ZvonimirMedak) [reply README.md:179]: I agree, we can create multiple docs files, and divide the whole Readme file 👌

File: Sentinel/Classes/Core/PreferencesInfo/Items/ToggleToolItem.swift (hunk 2 above)
Comment (ZvonimirMedak) [reply Sentinel/Classes/Core/PreferencesInfo/Items/ToggleToolItem.swift:11]: Great idea, I would also like to add the typealias so that we don't introduce breaking changes in the next release.

File: Sentinel/Classes/Core/PreferencesInfo/Views/OptionToggleView.swift (hunk 4 above)
Comment (ZvonimirMedak) [reply Sentinel/Classes/Core/PreferencesInfo/Views/OptionToggleView.swift:41]: I'm not really sure why I added that guard since SwiftUI should handle the updates properly. The guard is not needed here since the getter doesn't return an optional value. So we can reduce it to just `value = getter()`

**Comment by ZvonimirMedak** [comment]
Great comments @nikolamajcen, I fixed the mentioned improvements so please take a look at it again when you get a chance 😄

File: Sentinel/Classes/Core/PreferencesInfo/Items/PreferenceToggleItem.swift [hunk 9]
```diff
@@ -0,0 +1,76 @@
+//
+//  PreferenceToggleItem.swift
+//  Sentinel
+//
+//  Created by Zvonimir Medak on 17.03.2025..
+//
+
+import Foundation
+
+@available(*, deprecated, renamed: "PreferenceToggleItem", message: "ToggleToolItem has been renamed to PreferenceToggleItem, and please use the new name, as it is more consistent with other preference items")
```
Comment (nikolamajcen) [inline Sentinel/Classes/Core/PreferencesInfo/Items/PreferenceToggleItem.swift:10]: This part can be removed.
> and please use the new name, as it is more consistent with other preference items

I would rather add something like; Usage of the `ToggleToolItem` is not recommended due to the plan for deprecation.

**Review by nikolamajcen — APPROVED** [review]
Amazing work @ZvonimirMedak 👏 I've added one small comment, but overall, this is good to go!
